SAT/SATInference.cpp

```diff
@@ -24,15 +24,6 @@ namespace SAT
 // SATInference
 //
 
-/**
- * Collect first-order premises of @c cl into @c res. Make sure that elements in @c res are unique.
- */
-void SATInference::collectFOPremises(SATClause* cl, Stack<Unit*>& acc)
```

Comment:
Didn't you find the hardwired trivial lambda a nice convenience?

If this is for efficiency, maybe `collectFOPremises` could still live in the header?